pymongo/asynchronous/srv_resolver.py

```diff
@@ -92,12 +92,14 @@ def __init__(
             pass
 
         try:
-            self.__plist = self.__fqdn.split(".")[1:]
+            self.__plist = (
+                self.__fqdn.split(".")[1:]
+                if len(self.__fqdn.split(".")) > 2
+                else self.__fqdn.split(".")
+            )
```

Comment:
Can you refactor this so we only call split() once?

Reply:
Done in b6920f1110b3602d9525232169050fb39ed7baf1